### DISCUSSION TITLE
Distribute Sunshine under AGPL

### DISCUSSION THREAD

**Discussion by cathyjf** [LizardByte/community#540] [Sunshine]
### Select Topic Area

General

### Body

In LizardByte/Sunshine#150, there was a valiant effort to re-license Sunshine under AGPL v3, which was ultimately aborted because two possible authors of the work did not provide consent.

As it turns out, however, consent of those two authors is not necessary to distribute Sunshine primarily under AGPLv3. This is because section 13 of the current license (the GPL v3) explicitly allows combining works licensed under the GPLv3 with works licensed under the AGPLv3:

```
Notwithstanding any other provision of this License, you have
permission to link or combine any covered work with a work licensed
under version 3 of the GNU Affero General Public License into a single
combined work, and to convey the resulting work.  The terms of this
License will continue to apply to the part which is the covered work,
but the special requirements of the GNU Affero General Public License,
section 13, concerning interaction through a network will apply to the
combination as such.
```

So, almost all of the code can be relicensed under AGPLv3 (since the authors to most of the code consented), and then that code can be combined with the GPLv3 code, and the "combination" (including the GPLv3-only part) will be subject to the "the special requirements of the GNU Affero General Public License". This will accomplish the goal of LizardByte/Sunshine#150.

I think this path forward is desirable, and it can be implemented with a pull request that updates the licensing information based on the consents already given in LizardByte/Sunshine#150 combined with the permissions given to us by section 13 of the GPLv3.

**Comment by ReenigneArcher** [comment]
From my understanding, this would require dual licensing files? e.g. adding a license text to the top of each individual file?

**Comment by cathyjf** [comment]
It depends on the file. For files where all authors consent to relicense as AGPLv3, that file could be licensed under AGPLv3 only.

[a comment by an automated account is not shown]
